aten/src/ATen/native/cuda/Math.cuh

```diff
@@ -299,14 +407,100 @@ static inline C10_HOST_DEVICE scalar_t calc_i0e(scalar_t _x) {
   accscalar_t x = ::abs(static_cast<accscalar_t>(_x));
 
   if (x <= 8.0) {
-    const auto A = chebyshev_coefficients_A<accscalar_t>();
+    const auto A = chebyshev_coefficients_i0e_A<accscalar_t>();
     accscalar_t y = static_cast<accscalar_t>((x / 2.0) - 2.0);
     return static_cast<scalar_t>(chbevl(y, A, 30));
   }
 
-  const auto B = chebyshev_coefficients_B<accscalar_t>();
+  const auto B = chebyshev_coefficients_i0e_B<accscalar_t>();
   return static_cast<scalar_t>(chbevl(static_cast<accscalar_t>(32.0 / x - 2.0), B, 25) / ::sqrt(x));
 }
 
+template <typename scalar_t>
```

Comment:
These functions are identical to their CPU counterparts, right?

Reply:
Even with upcasting for the computation and then downcasting to original dtype has issue (mostly due to the way inherent characteristics of these dtypes). 
I think it would be better to support them if there is demand for the same.
```python
import torch
import scipy.special

t = torch.tensor(8.3750, dtype=torch.bfloat16)
print("TORCH:", torch.special.i1(t))
print("SCIPY:", scipy.special.i1(t.to(torch.float).numpy()))
print("BFLOAT16 CONVERSION:", torch.tensor(570.0086, dtype=torch.bfloat16))

expected = torch.special.i1(t.to(torch.double))
actual = torch.special.i1(t)

abs_err = torch.abs(expected - actual)
print("ABSOLUTE ERROR:", abs_err)
print("RELATIVE ERROR:", abs_err/expected)
```
Out:
```
TORCH: tensor(572., dtype=torch.bfloat16)
SCIPY: 570.0086
BFLOAT16 CONVERSION: tensor(572., dtype=torch.bfloat16)
ABSOLUTE ERROR: tensor(1.9914, dtype=torch.float64)
RELATIVE ERROR: tensor(0.0035, dtype=torch.float64)
```